js/CADWorker/CascadeStudioStandardUtils.js

```diff
@@ -1,19 +1,34 @@
+import {
+  oc,
+  GUIState,
+  opNumber,
+  setOpNumber,
+  setCurrentLineNumber,
+  argCache,
+  setCurrentOp,
+  usedHashes
+} from "./CascadeStudioWorkerState";
 // Miscellaneous Helper Functions used in the Standard Library
 
-// Caching functions to speed up evaluation of slow redundant operations
-var argCache = {}; var usedHashes = {}; var opNumber = 0; var currentOp = ''; var currentLineNumber = 0;
+function getCalleeName(fn) {
+  var ret = fn.toString();
+  ret = ret.substr("function ".length);
+  ret = ret.substr(0, ret.indexOf("("));
+  return ret;
+}
 
 /** Hashes input arguments and checks the cache for that hash.  
  * It returns a copy of the cached object if it exists, but will 
  * call the `cacheMiss()` callback otherwise. The result will be 
  * added to the cache if `GUIState["Cache?"]` is true. */
-function CacheOp(args, cacheMiss) {
+export function CacheOp(callee, args, cacheMiss) {
```

Comment:
This is what I've had to do to get around strick mode. here the function calling CacheOp has to pass in itself plus explicitly pass in its arguments. See this line https://github.com/zalo/CascadeStudio/pull/58/files#diff-a5196c193493e03c057c65c553d7b023601fb1f1ef472e47b68825c0fa2b6804R32

Reply:
I made another commit today (65da625) related to this as I noticed a problem that only presents itself when deployed. 

Basically the "in progress" console log that shows what operation is currently happening "Union", "Difference" etc was only showing single letters, and it because webpack's minification changes function names to single characters to save on the size of the js sent to the browser. and since we're passing the function directly to get its name at run time it suffers from this. I fixed it by changing an option of `keep_fnames` alternatively we could just use a string for the operation name instead actually passing the functions through to `CacheOp`.